add_me_to_your_PATH/Dockerfile.template

```diff
@@ -5,15 +5,24 @@ ARG CUDA_VERSION
 
 FROM julia:${JULIA_VERSION}-buster as pinned-extractor
 
-COPY *Project.toml *Manifest.toml .
+COPY Project.toml Manifest.toml Manifest.toml.julia_pod .
 
 # Remove all packages that are not pinned recursively from the manifest
 # (stdlibs count as pinned)
 RUN julia --project=. -e 'using Pkg;\
-                          unpinned = [p.name for p in values(Pkg.dependencies()) \
-                                      if p.is_direct_dep && !p.is_pinned && \
-                                      p.name ∉ values(Pkg.Types.stdlibs())]; \
-                          isempty(unpinned) || Pkg.rm(unpinned; mode=Pkg.PKGMODE_MANIFEST)'
+                          unregistered = [p.name for p in values(Pkg.dependencies()) \
+                                          if !p.is_tracking_registry]; \
+                          registered = [p.name for p in values(Pkg.dependencies()) \
+                                        if p.is_tracking_registry]; \
+                          previous = [p.name for p in values(Pkg.Types.read_manifest("Manifest.toml.julia_pod"))]; \
```

Comment:
This might break depending on the version of Julia. Might be better to use TOML stdlib and parse the manifest.

Reply:
Ok, I've changed it so that instead of saving a copy of `Manifest.toml` to `Manifest.toml.julia_pod`, I save a list of sysimage package names to `julia_pod/sysimage.packages`, one package per line. This is less confusing since I was only using the package names from `Manifest.toml.julia_pod`.

So reading in the list of deps is just `readlines("julia_pod/sysimage.packages")`.